File: 1436 destination-city/solution.java
class Solution {
    public String destCity(List<List<String>> paths) {
        Set<String> s = new HashSet<>();
        for(List<String> l : paths) s.add(l.get(0));
        for(List<String> l:paths) {
            if(!s.contains(l.get(1))) return l.get(1);
        }
        return "";
    }
}

File: 1436 destination-city/solution.rs
use std::collections::HashSet;
impl Solution {
    pub fn dest_city(paths: Vec<Vec<String>>) -> String {
         let mut s = HashSet::new();
         for p in &paths {
            s.insert(&p[0]);
         }
         for p in &paths {
            if !s.contains(&p[1]) {
                return p[1].clone();
            }
         }
         "".to_string()
    }
}
